src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/armor/StomperBoots.java

```diff
@@ -37,49 +39,49 @@ public StomperBoots(Category category, SlimefunItemStack item, RecipeType recipe
 
     /**
      * This will apply the "stomp" effect to the given {@link EntityDamageEvent}.
-     * 
+     *
      * @param fallDamageEvent
      *            The {@link EntityDamageEvent} in which the {@link Player} has taken fall damage
      */
     public void stomp(EntityDamageEvent fallDamageEvent) {
-        Player p = (Player) fallDamageEvent.getEntity();
-        p.getWorld().playSound(p.getLocation(), Sound.ENTITY_ZOMBIE_BREAK_WOODEN_DOOR, 1F, 2F);
-        p.setVelocity(new Vector(0, 0.7, 0));
+        Player player = (Player) fallDamageEvent.getEntity();
+        player.getWorld().playSound(player.getLocation(), Sound.ENTITY_ZOMBIE_BREAK_WOODEN_DOOR, 1F, 2F);
+        player.setVelocity(new Vector(0, 0.7, 0));
 
-        for (Entity n : p.getNearbyEntities(4, 4, 4)) {
-            if (n instanceof LivingEntity && n.isValid() && !n.getUniqueId().equals(p.getUniqueId())) {
-                Vector velocity = getShockwave(p.getLocation(), n.getLocation());
-                n.setVelocity(velocity);
+        for (Entity entity : player.getNearbyEntities(4, 4, 4)) {
+            if (entity instanceof LivingEntity && canMoveEntity((LivingEntity) entity, player)) {
+                Vector velocity = getShockwave(player.getLocation(), entity.getLocation());
+                entity.setVelocity(velocity);
 
                 // Check if it's not a Player or if PvP is enabled
-                if (!(n instanceof Player) || (p.getWorld().getPVP() && SlimefunPlugin.getProtectionManager().hasPermission(p, n.getLocation(), ProtectableAction.ATTACK_PLAYER))) {
-                    EntityDamageByEntityEvent event = new EntityDamageByEntityEvent(p, n, DamageCause.ENTITY_ATTACK, fallDamageEvent.getDamage() / 2);
+                if (!(entity instanceof Player) || (player.getWorld().getPVP() && SlimefunPlugin.getProtectionManager().hasPermission(player, entity.getLocation(), ProtectableAction.ATTACK_PLAYER))) {
+                    EntityDamageByEntityEvent event = new EntityDamageByEntityEvent(player, entity, DamageCause.ENTITY_ATTACK, fallDamageEvent.getDamage() / 2);
                     Bukkit.getPluginManager().callEvent(event);
 
                     if (!event.isCancelled()) {
-                        ((LivingEntity) n).damage(event.getDamage());
+                        ((LivingEntity) entity).damage(event.getDamage());
                     }
                 }
             }
         }
 
         for (BlockFace face : BlockFace.values()) {
-            Block b = p.getLocation().getBlock().getRelative(BlockFace.DOWN).getRelative(face);
-            p.getWorld().playEffect(b.getLocation(), Effect.STEP_SOUND, b.getType());
+            Block block = player.getLocation().getBlock().getRelative(BlockFace.DOWN).getRelative(face);
+            player.getWorld().playEffect(block.getLocation(), Effect.STEP_SOUND, block.getType());
         }
```

Comment:
I don't really see much of a problem with the naming of p, n and b but sure.

Reply:
p and b could be used fine like that but n for entity is a bit odd here, and besides I think using full names is cleaner and more easily readable